packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -232,6 +295,66 @@ export abstract class BaseService extends Resource
     return ret;
   }
 
+  public loadBalancerTarget(options: LoadBalancerTargetOptions): IEcsLoadBalancerTarget {
+    const self = this;
+    const hostPort = this.taskDefinition._findHostPort(options);
+    return {
+      attachToApplicationTargetGroup(targetGroup: elbv2.ApplicationTargetGroup): elbv2.LoadBalancerTargetProps {
+        let portRange: ec2.Port;
+        if (hostPort.protocol === Protocol.TCP) {
+          portRange = hostPort.port === 0 ? EPHEMERAL_PORT_RANGE : ec2.Port.tcp(hostPort.port);
+        } else {
+          portRange = hostPort.port === 0 ? EPHEMERAL_PORT_RANGE : ec2.Port.udp(hostPort.port);
+        }
+        targetGroup.registerConnectable(self, portRange);
+        return self.attachToELBv2(targetGroup, options);
+      },
+
+      attachToNetworkTargetGroup(targetGroup: elbv2.NetworkTargetGroup): elbv2.LoadBalancerTargetProps {
+        return self.attachToELBv2(targetGroup, options);
+      },
+    };
+  }
+
+  public registerContainerTargets(containers: TargetProps[]) {
+    for (const container of containers) {
+      for (const containerTarget of container.containerTargets) {
+        const target = {
+          containerName: container.containerName,
+          containerPort: containerTarget.containerPort,
+          protocol: containerTarget.protocol
+        };
+        for (const targetGroupProp of containerTarget.targetGroups) {
+          if (targetGroupProp.listener instanceof elbv2.ApplicationListener) {
```

Comment:
Not a fan of all the type juggling that is happening here. I don't really see the need for this function, and the implementation scares me a little.